### PR TITLE
Jax Criteo1TB DLRM-Small Workload

### PR DESCRIPTION
**Pull request by znado**
I haven't confirmed that this gets the proper validation target value yet, but this passes tests/reference_submission_tests.py. In a follow-up PR I can add the data downloading/prep script, and confirm that it gets the correct values.

We also need to finalize the validation split, which we have someone working on after the NeurIPS deadline.

### CODE REVIEW

File: algorithmic_efficiency/workloads/criteo1tb/input_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    ds = ds.repeat()
+  ds = ds.interleave(
+      tf.data.TextLineDataset,
+      cycle_length=128,
+      block_length=per_device_batch_size // 8,
+      num_parallel_calls=128,
+      deterministic=False)
+  ds = ds.batch(
+      per_device_batch_size, drop_remainder=True
+  )  # DO NOT SUBMIT note that we will select a validation split size that is evenly divisible
```
Comment (fsschneider) [inline algorithmic_efficiency/workloads/criteo1tb/input_pipeline.py:67]: Should this be changed to a `TODO`?

File: algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+  def num_train_examples(self):
+    return 4_195_197_692
+
+  @property
+  def num_eval_train_examples(self):
+    return 100_000
+
+  @property
+  def num_validation_examples(self):
+    return 89_137_318  # DO NOT SUBMIT
```
Comment (fsschneider) [inline algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py:65]: Probably also a `TODO`

File: algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+  def train_mean(self):
+    return 0.0
+
+  @property
+  def train_stddev(self):
+    return 1.0
+
+  @property
+  def max_allowed_runtime_sec(self):
+    return 8 * 60 * 60  # 8 hours. # DO NOT SUBMIT
```
Comment (fsschneider) [inline algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py:81]: I believe, in our meeting, we decided to use 6 hours as the maximum runtime. But we might want to iterate on this anyway after seeing some experimental results.

File: algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+  @property
+  def eval_period_time_sec(self):
+    return 20 * 60  # 20 minutes. # DO NOT SUBMIT
```
Comment (fsschneider) [inline algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py:85]: Probably also a `TODO`

File: algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+    del rng
+    del update_batch_norm
+    inputs = augmented_and_preprocessed_input_batch['inputs']
+    targets = augmented_and_preprocessed_input_batch['targets']
+    logits_batch = self._flax_module.apply({'params': params}, inputs, targets)
+    return logits_batch, None
+
+  @property
+  def step_hint(self):
+    return 64_000  # DO NOT SUBMIT
```
Comment (fsschneider) [inline algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py:229]: Probably also a `TODO`

**Review by fsschneider — APPROVED** [review]
LGTM!

File: algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py (hunk 3 above)
Comment (znado) [reply algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py:81]: SGTM, done.

File: algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py (hunk 4 above)
Comment (znado) [reply algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py:85]: Done.

File: algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py (hunk 5 above)
Comment (znado) [reply algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py:229]: Done (I had a commit that changed these that I meant to push before emailing you, but forgot to hit "enter" on the git push command :upside_down_face: )

File: setup.cfg [hunk 6]
```diff
@@ -76,7 +76,7 @@ dev =
 gnn =
```
[the remark this answers is not in the document]
Comment (znado) [reply setup.cfg:82]: done!

**Pull request merged**